lib/lines.ts

```diff
@@ -656,6 +699,10 @@ Lp.slice = function(start, end) {
     end = this.lastPos();
   }
 
+  if (!start) {
+    throw new Error("cannot slice with end but not start");
+  }
```

Comment:
Needed this type guard in order for TS to know `start` is present later on in this function